spaceship/lib/spaceship/tunes/application.rb

```diff
@@ -120,11 +130,48 @@ def resolution_center
         client.get_resolution_center(apple_id, platform)
       end
 
+      def platforms
+        platforms = []
+        version_sets.each do |version_set|
+          platforms << version_set.platform
+        end
+        platforms
+      end
+
+      def types
+        types = []
+        version_sets.each do |version_set|
+          types << version_set.type
+        end
+        types
+      end
+
+      def type
+        if self.version_sets.nil?
+          raise 'The application has no version sets and Spaceship does not know what to do here.'
+        end
+
+        if self.version_sets.length == 1
+          version_sets[0].platform
+        end
+        platform = Spaceship::Tunes::AppVersionCommon.find_platform(raw_data['versionSets'])
+        platform['type']
+      end
+
       # kept for backward compatibility
       # tries to guess the platform of the currently submitted apps
       # note that as ITC now supports multiple app types, this might break
       # if your app supports more than one
       def platform
+        if self.version_sets.nil?
+          raise 'The application has no version sets and Spaceship does not know what to do here.'
+        end
+
+        if self.version_sets.length == 1
+          version_sets[0].platform
+        elsif self.platforms == %w(ios appletvos)
+          'ios'
+        end
         Spaceship::Tunes::AppVersionCommon.find_platform(raw_data['versionSets'])['platformString']
```

Comment:
@mpirri  If I remember correctly version_sets did not have the right platformString set.